x-pack/plugins/security_solution/scripts/endpoint/common/fleet_services.ts

```diff
@@ -223,7 +225,15 @@ export const getAgentVersionMatchingCurrentStack = async (
   kbnClient: KbnClient
 ): Promise<string> => {
   const kbnStatus = await kbnClient.status.get();
-  let version = kbnStatus.version.number;
+  const agentVersions = await axios
+    .get('https://artifacts-api.elastic.co/v1/versions')
+    .then((response) =>
+      filter(response.data.versions, (versionString) => !versionString.includes('SNAPSHOT'))
+    );
+
+  let version =
+    semver.maxSatisfying(agentVersions, `<=${kbnStatus.version.number}`) ??
+    kbnStatus.version.number;
```

Comment:
As you have mentioned earlier, in case the new version of the stack is branched out, the kibana version is being bumped up (currently it's 8.8.0), but it takes some time before the agent docker image is available (like now, the latest available agent version is 8.7.0).
To make sure the script works in that case as well, I've added a call to API to get the available agent versions and then I'm trying to find the highest agent version that works with the stack version, so in that case, the function will return `8.7.0`.
In case someone would try to run the script from the `8.5` branch, because this version is not available in API results anymore, it will fallback to `kbnStatus.version.number`, because the docker image should be still available for the past versions

Reply:
Ok. thats what I thought. But this method here is clear in that it only meant to return the version matching the current stack. It's not meant to validate that a download exists for the given stack number - that IMO should be a different method. Your change could also confuse users (ex. running the CLI tool) that might have asked to run a specific version and then end up finding out that they are actually running a different one. 

Can I suggest that you create a specific method for this Cypress use case and use that instead?  Something like:

```
/**
 * Returns the Agent version that is available for install (will check `artifacts-api.elastic.co/v1/versions`)
 * that is equal to or less than `maxVersion`.
 * @param maxVersion
 */
const getLatestAvailableAgentVersion = async (maxVersion: string): Promise<string> => {
  const agentVersions = await axios
    .get('https://artifacts-api.elastic.co/v1/versions')
    .then((response) =>
      filter(response.data.versions, (versionString) => !versionString.includes('SNAPSHOT'))
    );

  const version = semver.maxSatisfying(agentVersions, `<=${maxVersion}`);

  if (!version) {
    throw new Error(`No available version found that are equal to or less than ${maxVersion}`);
  }

  return version;
};
```

Not sure if we need this method in this module here under `script/common`. Maybe it should be private to the cypress code under `x-pack/test/defend_workflows_cypress/**`